LookupSystemService/Controllers/UserController.cs

```diff
@@ -12,19 +17,113 @@ namespace LookupSystemService.Controllers
     [Route("[controller]")]
     public class UserController : ControllerBase
     {
+
         private readonly ILogger<UserController> _logger;
+        private readonly IRepository<User> userRepo;
 
-        public UserController(ILogger<UserController> logger)
+        public UserController(ILogger<UserController> logger, LookupSystemDbContext contex)
         {
             _logger = logger;
+            userRepo = new UserRepository(contex);
+        }
+
+        [HttpGet("GetUserByEmail/{email}")]
+        public IEnumerable<EmailSearchUserViewModel> GetUserByEmail(string email)
+        {
+            try
+            {
+                var config = new MapperConfiguration(cfg => cfg.CreateMap<User, EmailSearchUserViewModel>()
+                .ForMember("FirstName", opt => opt.MapFrom(m => m.UserContact.FirstName))
+                .ForMember("LastName", opt => opt.MapFrom(m => m.UserContact.LastName))
+                .ForMember("Email", opt => opt.MapFrom(m => m.UserContact.Email)));
+
+                var mapper = new Mapper(config);
+                var users = mapper.Map<List<EmailSearchUserViewModel>>(userRepo.GetUserByEmail(email));
```

Comment:
review automapper extension for EF and use it. Additional questions could be asked directly